sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -764,7 +764,16 @@ ProgramManager::getDeviceImage(OSModuleHandle M, KernelSetId KSId,
     debugPrintBinaryImages();
   }
   std::lock_guard<std::mutex> Guard(Sync::getGlobalLock());
-  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *m_DeviceImages[KSId];
+  std::vector<RTDeviceBinaryImageUPtr> *ImgsPtr = nullptr;
+  {
+    auto It = m_DeviceImages.find(KSId);
+    if (It == m_DeviceImages.end())
+      throw runtime_error(
+          "No device image found for KernelSetId " + std::to_string(KSId) + " was found",
+          PI_INVALID_KERNEL_NAME);
+    ImgsPtr = It->second.get();
+  }
+  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *ImgsPtr;
```

Comment:
Looks good to me.
Minor comment: the additional braces on the lines 768 and 775 seem redundant and the code could be written a bit shorter:
```suggestion
  auto It = m_DeviceImages.find(KSId);
  if (It == m_DeviceImages.end())
    throw runtime_error(
        "No device image found for KernelSetId " + std::to_string(KSId) + " was found",
        PI_INVALID_KERNEL_NAME);
  std::vector<RTDeviceBinaryImageUPtr> &Imgs = *It->second.get();
```

Reply:
Fixed.